cluster-autoscaler/expander/nodecountbalance/nodecountbalance.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodecountbalance
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+type nodecountbalance struct {
+}
+
+// NewStrategy returns an expansion strategy that randomly picks between node groups
+func NewStrategy() expander.Strategy {
+	return &nodecountbalance{}
+}
+
+// BestOption selects from the expansion options to balance node groups in their node counts
+func (r *nodecountbalance) BestOption(expansionOptions []expander.Option, nodeInfo map[string]*schedulercache.NodeInfo) *expander.Option {
+	pos := 0
+	for i, o := range expansionOptions {
+		if i == 0 {
+			continue
+		}
+		if o.NodeCount < expansionOptions[pos].NodeCount {
```

Comment:
@mumoshu I'm thinking to just take a node from each node group and compare them to find groups with similar nodes. By similar I mean basically identical except for a pre-determined set of things like zone label.

I will certainly add a flag to enable / disable the whole balancing logic. I see a potential benefit of allowing explicit `--node-group-family` flag to override the default logic, but I'm not convinced we really need it and it adds complexity. So I'm leaning towards not adding it, at least unless someone asks for it. But I don't really have a strong opinion here.

Unfortunately adding the error counting metric turned out to be more complex than I initially expected, so I will start working on zone balancing next week. Sorry about that.